crates/tinymist/src/state.rs

```diff
@@ -30,17 +30,13 @@ impl CompileServer {
 impl TypstLanguageServer {
     /// Pin the entry to the given path
     pub fn pin_entry(&mut self, new_entry: Option<ImmutPath>) -> Result<(), Error> {
-        self.pinning = new_entry.is_some();
```

Comment:
I see `self.pinning` is not updated now, so we need to update `focus_entry` and other related functions as well in this PR.